tinyiiod.c

```diff
@@ -164,6 +174,40 @@ void tinyiiod_do_close(struct tinyiiod *iiod, const char *device)
 	tinyiiod_write_value(iiod, ret);
 }
 
+int32_t tinyiiod_do_close_instance(struct tinyiiod *iiod)
+{
+	return iiod->ops->close_instance();
+}
+
+int32_t tinyiiod_do_writebuf(struct tinyiiod *iiod,
```

Comment:
this doesn't seem to differ much from `tinyiiod_do_write_attr()` in terms of behavior;

if `tinyiiod_read_nonblocking()` is non-blocking, then you could also just get a "no-data-yet" type of error and things fail/error out;

if you do need to implement a `tinyiiod_read_wait()` then you don't need to make this nonblocking

if you extend the buffer in `tinyiiod_do_write_attr()` you pretty much make it identical with this function

Reply:
tinyiiod_do_writebuf(), has a tinyiiod_write_value(iiod, (int) bytes_count);, which is a kind of "ack", after this point, client sends data. If the server is not prepared, for receiving of the data (which tinyiiod_read_nonblocking() call does), some bytes are lost. This solution was adopted by me, after a lot of iterations. You can try for better results, until then, I will suggest this one.